src/atomate2/common/flows/mpmorph.py

```diff
@@ -0,0 +1,493 @@
+"""Define code agnostic MPMorph flows.
+
+This file generalizes the MPMorph workflows of
+https://github.com/materialsproject/mpmorph
+originally written in atomate for VASP only to a more general
+code agnostic form.
+
+For information about the current flows, contact:
+- Bryant Li (@BryantLi-BLI)
+- Aaron Kaplan (@esoteric-ephemera)
+- Max Gallant (@mcgalcode)
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Literal
+
+import numpy as np
+from jobflow import Flow, Maker, Response, job
+
+from atomate2.common.jobs.eos import MPMorphPVPostProcess, _apply_strain_to_structure
+
+if TYPE_CHECKING:
+    from pathlib import Path
+    from typing import Any
+
+    from jobflow import Job
+    from pymatgen.core import Structure
+    from typing_extensions import Self
+
+    from atomate2.common.jobs.eos import EOSPostProcessor
+
+
+@dataclass
+class EquilibriumVolumeMaker(Maker):
+    """
+    Equilibrate structure using NVT + EOS fitting.
+
+    Parameters
+    ----------
+    name : str = "Equilibrium Volume Maker"
+        Name of the flow
+    md_maker : Maker
+        Maker to perform NVT MD runs
+    postprocessor : atomate2.common.jobs.eos.EOSPostProcessor
+        Postprocessing step to fit the EOS
+    initial_strain : float | tuple[float,float] = 0.2
+        Initial percentage linear strain to the apply to the structure
+    min_strain : float, default = 0.5
+        Minimum absolute percentage linear strain to apply to the structure
+    max_attempts : int | None = 20
+        Number of times to continue attempting to equilibrate the structure.
+        If None, the workflow will not terminate if an equilibrated structure
+        cannot be determined.
+    """
+
+    name: str = "Equilibrium Volume Maker"
+    md_maker: Maker | None = None
+    postprocessor: EOSPostProcessor = field(default_factory=MPMorphPVPostProcess)
+    initial_strain: float | tuple[float, float] = 0.2
+    min_strain: float = 0.5
+    max_attempts: int | None = 20
+
+    @job
+    def make(
+        self,
+        structure: Structure,
+        prev_dir: str | Path | None = None,
+        working_outputs: dict[str, Any] | None = None,
+    ) -> Flow:
+        """
+        Run an NVT+EOS equilibration flow.
+
+        Parameters
+        ----------
+        structure : Structure
+            structure to equilibrate
+        prev_dir : str | Path | None (default)
+            path to copy files from
+        working_outputs : dict or None
+            contains the outputs of the flow as it recursively updates
+
+        Returns
+        -------
+        .Flow, an MPMorph flow
+        """
+        if working_outputs is None:
+            if isinstance(self.initial_strain, float | int):
```

Comment:
this `make` looks like it could benefit from more checking of assumptions and raising helpful error messages if they fail.

e.g. 

```
       if isinstance(self.initial_strain, float | int):
          self.initial_strain = (
              -abs(self.initial_strain),
              abs(self.initial_strain),
          )
      elif not isinstance(self.initial_strain, tuple) or len(self.initial_strain) != 2:
         raise ValueError(f"initial_strain must be float | tuple[float, float], got {initial_strain}")
```